git_wrapper/repo.py

```diff
@@ -0,0 +1,185 @@
+#! /usr/bin/env python
+"""This module acts as an interface for common git tasks"""
+
+import logging
+import os
+import shutil
+
+import git
+from future.utils import raise_from
+
+from git_wrapper.branch import GitBranch
+from git_wrapper import exceptions
+from git_wrapper.remote import GitRemote
+
+
+class GitRepo(object):
```

Comment:
I wonder if we should consider using this class more like a proxy to access things like branches, remotes, commits etc instead of using it a as a base.

Reply:
No sure what that would look like at this point. Let see where this redesign takes us.